### DISCUSSION TITLE
Interpolation function

### DISCUSSION THREAD

**Discussion by GrblGru** [grblHAL/core#210] [Q&A]
If you export geometries (arcs, ellipses, Bezier curves) from a DXF editor as DXF, you very often get polygons consisting of relatively large straight line pieces. Even if you create a nice circle or arc, if you project it onto an inclined plane, you get a point cloud, where the points are again connected with straight line segments.

If you move these polygons as GCode with a constant feedrate, you often get feedrate fluctuations because the controller tries to move the points precisely enough. In addition, you can often see the individual facets on the surface, which is very unattractive.

I would like to see a function that interpolates additional points between the points and moves them at a constant feed rate. Is there a function in GrblHAL that does something similar ? Is 'G5 cubic spline' suitable for this ? I haven't found anything about it in the documentation yet.

**Comment by terjeio** [comment]
> I would like to see a function that interpolates additional points between the points and moves them at a constant feed rate. Is there a function in GrblHAL that does something similar ?

G5 and G5.1 does that in a similar way as G2 and G3 does. G5.2/G5.3 is not supported and I have no immediate plan to change that.

BTW increasing the planner buffer size may help to increase the feed rate that can be achieved.

> I haven't found anything about it in the documentation yet.

Non NIST G- and M-codes supported by grblHAL generally follow LinuxCNC specifications, G5 and G5.1 is [documented here](http://linuxcnc.org/docs/html/gcode/g-code.html#gcode:g5).

A bit more info and example code can also be found in [these issues](https://github.com/search?q=org%3AgrblHAL+spline&type=issues).

**Comment by GrblGru** [comment]
Sorry for the many questions but GrblHAL is still very new to me. The tip with the Linux documentation is very valuable. G5 sounds interesting. I will give it a try.
Thanks for your help.  :)

**Comment by GrblGru** [comment]
I have now greatly reduced the point spacing in my program and made use of the max. Planner buffer size.
With this, even fast movements can be executed without any fluctuations. This was not possible with the other controllers so far. 
Great :)
